ladybug/header.py

```diff
@@ -129,6 +129,18 @@ def ToString(self):
         """Overwrite .NET ToString."""
         return self.__repr__()
 
+    def __key(self):
+        return (self.data_type, self.unit, self.analysis_period, self.metadata)
+
+    def __hash__(self):
+        return hash(self.__key())
```

Comment:
This is also going to fail since you aren't hashing the `data_type`, the `analysis_period` in the key. Nor are you converting the `metadata` into an immutable format. So Just remove this for now unless you are want to implement it correctly.